### ISSUE TITLE
NIL is not a TEdit document error when renaming NoteCards card

### ISSUE THREAD

**Issue by pamoroso** [Interlisp/medley#1867]
**Describe the bug**

Renaming a NoteCards card with the `Assign Title` command yields a break window with the error:

```
In ERROR:
NIL is not a Tedit document
```

**To Reproduce**

1. launch Medley
2. load NoteCards
3. in the file browser, open the notefile `DEMO.NOTEFILE`
4. open the card `How to use this demo`
5. from the left-click title bar menu of the card, select `Assign Title`
6. enter a new title or just press `ENTER`

**Expected behavior**

The card title is updated with no errors.

**Screenshots**

The break window:

![notecards-card-title-error](https://github.com/user-attachments/assets/7ede549a-2006-44d2-8be6-7256db7b9e2f)

The full desktop with the break window:

![notecards-card-title-error-desktop](https://github.com/user-attachments/assets/2af2d2b4-d14a-453c-86bd-d4c64b32ba4e)

**Context (please complete the following information):**

 - OS: Linux Mint Cinnamon
 - OS Version: 22
 - Host arch: x86_64 (System76 Merkaat)
 - Maiko version: N/A
 - IL:MAKESYSDATE: 27-Oct-2024 11:48:39

**Additional context**

The issue may be related to #1829.

The `BTV!` output:

```
NC.TEditMenuFn/5(debug)BTV!


Basic frame at   40430
  40422:       0      0 MESS1 NIL
  40424:     150  11350 MESS2 "is not a Tedit document"
  40426:       0      0 NOBREAK NIL
  40430:  100000  40422 

Frame xtn at   40432, frame name= ERROR
  40432:  140005  40403 [USE= 5, X,  alink]
  40434:     164 117600 [fn header]
  40436:   40456    170 [next, pc]
  40440:  177777 177777 [nametable]
  40442:   40430  40402 [blink, clink]
  40444:  177777 177777 [fvar HELPFLAG  not looked up]
  40446:  127034     54 [fvar *LAST-CONDITION* 
 non-stack binding]
  40450:  100000  40446 [padding]
  40452:     150  11350 [padding]
  40454:     153 136614 #<Condition INTERLISP-ERROR @ 153,136614>
ERROR


Basic frame at   40366
  40362:       0      0 TSTREAM? NIL
  40364:       0      0 NOERROR NIL
  40366:  100000  40362 

Frame xtn at   40370, frame name= TEXTSTREAM
  40370:  140001  40353 [USE= 1, X,  alink]
  40372:     146  73400 [fn header]
  40374:   40422    355 [next, pc]
  40376:     161 171450 [nametable]
  40400:   40366  40352 [blink, clink]
  40402:       0      0 TS NIL
  40404:       0      0 WINDOW NIL
  40406:       0      0 X NIL
  40410:       0      0 "local" NIL
  40412:       0      0 [padding]
  40414:       0      0 [padding]
  40416:  177773      6 
  40420:       0      0 NIL
TEXTSTREAM


Basic frame at   40336
  40332:       0      0 TEXTOBJ? NIL
  40334:       0      0 NOERROR NIL
  40336:  100000  40332 

Frame xtn at   40340, frame name= TEXTOBJ
  40340:  140001  40311 [USE= 1, X,  alink]
  40342:     146  76520 [fn header]
  40344:   40362    100 [next, pc]
  40346:       0      0 [nametable]
  40350:   40336  40310 [blink, clink]
  40352:  177777 177777 TSTRM [unbound]
  40354:  177777 177777 [padding]
  40356:  177775      2 [padding]
  40360:       0      0 [padding]
TEXTOBJ


Basic frame at   40274
  40264:       0      0 TEXTOBJ NIL
  40266:     132  31512 FN NC.TEditCollectObjectsA0001
  40270:       0      0 FNARG NIL
  40272:       0    114 COLLECT? T
  40274:  100000  40264 

Frame xtn at   40276, frame name= TEDIT.MAP.OBJECTS
  40276:  140001  40247 [USE= 1, X,  alink]
  40300:     146  64310 [fn header]
  40302:   40332    152 [next, pc]
  40304:     134  76420 [nametable]
  40306:   40274  40246 [blink, clink]
  40310:      16      1 CH# 1
  40312:       0      0 "local" NIL
  40314:       0      0 OBJ NIL
  40316:       0      0 FNVAL NIL
  40320:       0      0 PC NIL
  40322:  177777 177777 [padding]
  40324:     161 153620 [padding]
  40326:       0    114 [padding]
  40330:  177772     10 
TEDIT.MAP.OBJECTS


Basic frame at   40232
  40226:       0      0 TEXTSTREAM NIL
  40230:     132  31656 TESTFN 
NC.UpdateLinkImagesInTextA0001
  40232:  100000  40226 

Frame xtn at   40234, frame name= NC.TEditCollectObjects
  40234:  140001  40201 [USE= 1, X,  alink]
  40236:     132  27620 [fn header]
  40240:   40264    100 [next, pc]
  40242:     134  76704 [nametable]
  40244:   40232  40200 [blink, clink]
  40246:       0      0 "local" NIL
  40250:       0      0 "local" NIL
  40252:       0      0 "local" NIL
  40254:       0      0 "local" NIL
  40256:     161 171120 [padding]
  40260:     132  31656 [padding]
  40262:  177773      6 
NC.TEditCollectObjects


Basic frame at   40164
  40160:     122  54724 SourceCardOrWindow {CardObject}#
122,54724
  40162:     122  44064 DestinationCard {CardObject}#122
,44064
  40164:  100000  40160 

Frame xtn at   40166, frame name= 
NC.UpdateLinkImagesInText
  40166:  140001  40121 [USE= 1, X,  alink]
  40170:     132  30400 [fn header]
  40172:   40226    212 [next, pc]
  40174:     133  12060 [nametable]
  40176:   40164  40120 [blink, clink]
  40200:     161 152000 Window {WINDOW}#161,152000
  40202:       0      0 TextObject NIL
  40204:       0      0 LinksAndPointers NIL
  40206:       0      0 SourceCard NIL
  40210:       0      0 WasDirtyFlg NIL
  40212:  177777 177777 LinkOrPointer [unbound]
  40214:       0      0 [padding]
  40216:     122  44064 [padding]
  40220:  177772     10 
  40222:     161 152000 {WINDOW}#161,152000
  40224:       0      0 NIL
NC.UpdateLinkImagesInText


Basic frame at   40104
  40100:     122  54724 SourceWindowOrCard {CardObject}#
122,54724
  40102:     122  44064 DestinationCard {CardObject}#122
,44064
  40104:  100000  40100 

Frame xtn at   40106, frame name= NC.UpdateLinkImages
  40106:  140001  40015 [USE= 1, X,  alink]
  40110:     104 166734 [fn header]
  40112:   40160    575 [next, pc]
  40114:     122  44064 [nametable]
  40116:   40104  40014 [blink, clink]
  40120:     161 152000 SourceWindow {WINDOW}#161,152000
  40122:     122  54724 SourceCard {CardObject}#122,
54724
  40124:     135 155226 NoteCardType FileBox
  40126:       0      0 ShrunkenWin NIL
  40130:     132  31644 "local" 
NC.UpdateLinkImagesInText
  40132:  177777 177777 NoteCardType [unbound]
  40134:  177777 177777 CardTypeName [unbound]
  40136:  177777 177777 CardTypeRecord [unbound]
  40140:  177777 177777 FieldValue [unbound]
  40142:  177777 177777 [padding]
  40144:  100000  40142 [padding]
  40146:     122  44064 [padding]
  40150:  177772     10 
  40152:     135 155226 FileBox
  40154:       0      0 NIL
  40156:       0      0 NIL
NC.UpdateLinkImages


Basic frame at   40000
  37776:       0   1534 SI::*CLEANUP-FORMS* 
SI::RESETUNWIND
  40000:  100000  37776 

Frame xtn at   40002, frame name= SI::*UNWIND-PROTECT*
  40002:  140001  37727 [USE= 1, X,  alink]
  40004:     104 114160 [fn header]
  40006:   40100   1063 [next, pc]
  40010:   40036    152 [nametable]
  40012:   40000  37726 [blink, clink]
  40014:  177777 177777 \NF [unbound]
  40016:  177777 177777 local [unbound]
  40020:  177777 177777 \Cache [unbound]
  40022:     122  34732 LinksWereCachedFlg {LinksCache}#
122,34732
  40024:     122  41052 FromLink {Link}#122,41052
  40026:     122  54724 ContainingCard {CardObject}#122,
54724
  40030:       0      0 RemoteCrossFileLinkCard NIL
  40032:  177777 177777 ToLink [unbound]
  40034:  177777 177777 ContainingCard [unbound]
  40036:  177777 177777 RemoteCrossFileLinkCard
 [unbound]
  40040:   37656      1 [fvar Card  on stack]
  40042:   37636      1 [fvar NewTitle  on stack]
  40044:   37640      1 [fvar InterestedWindow 
 on stack]
  40046:   37746      1 [fvar SI::*RESETFORMS* 
 on stack]
  40050:   37660      1 [fvar OldTitle  on stack]
  40052:   37662      1 [fvar Window  on stack]
  40054:   37634      1 [fvar NoClearMsgFlg  on stack]
  40056:  177777 177777 [padding]
  40060:     134  13040 [padding]
  40062:       0   1534 [padding]
  40064:     132  12020 ((& NIL))
  40066:     141    460 "How to use this demo"
  40070:  177776      6 
  40072:     150  40646 ({Link}#122,41052)
  40074:  177776     10 
  40076:  177775     14 
SI::*UNWIND-PROTECT*


Basic frame at   37712
  37710:       0   1534 SI::*CLEANUP-FORMS* 
SI::RESETUNWIND
  37712:  100000  37710 

Frame xtn at   37714, frame name= SI::*UNWIND-PROTECT*
  37714:  140001  37657 [USE= 1, X,  alink]
  37716:     104 114700 [fn header]
  37720:   37776   1056 [next, pc]
  37722:       0      0 [nametable]
  37724:   37712  37656 [blink, clink]
  37726:       0      0 OpInProgress NIL
  37730:       0      0 "local" NIL
  37732:       0      0 "local" NIL
  37734:       0      0 "local" NIL
  37736:       0      0 "local" NIL
  37740:     133  75416 "local" 
NCLocalDevice.ObtainWritePermission
  37742:  177777 177777 \NF [unbound]
  37744:       0      0 LISPXHIST NIL
  37746:     132  12020 SI::*RESETFORMS* ((& NIL))
  37750:       0      0 RESETSTATE NIL
  37752:   37656      1 [fvar Card  on stack]
  37754:   37666      1 [fvar SI::*RESETFORMS* 
 on stack]
  37756:   37664      1 [fvar LISPXHIST  on stack]
  37760:  177777 177777 [padding]
  37762:  100000  37756 [padding]
  37764:       0   1534 [padding]
  37766:     174  22600 
#<Lock NoteCards Meta-lock/174,22600>
  37770:  177771     12 
  37772:       0      0 NIL
  37774:  177774     22 
SI::*UNWIND-PROTECT*


Basic frame at   37642
  37632:     122  23570 CardIdentifier {WINDOW}#122,
23570
  37634:       0      0 NoClearMsgFlg NIL
  37636:     141    460 NewTitle "How to use this demo"
  37640:     122  23570 InterestedWindow {WINDOW}#122,
23570
  37642:  100000  37632 

Frame xtn at   37644, frame name= NC.AssignTitle
  37644:  140001  37611 [USE= 1, X,  alink]
  37646:     133  50524 [fn header]
  37650:   37710    214 [next, pc]
  37652:     134  64346 [nametable]
  37654:   37642  37610 [blink, clink]
  37656:     122  44064 Card {CardObject}#122,44064
  37660:     141    740 OldTitle "How to use this demo"
  37662:     122  23570 Window {WINDOW}#122,23570
  37664:       0      0 LISPXHIST NIL
  37666:     132  12050 SI::*RESETFORMS* ((& &) (& &) (&
 "Assign Title") (& #<Process NC.TEditMenuFn/122,62204>)
)
  37670:       0      0 RESETSTATE NIL
  37672:   34412     54 [fvar LISPXHIST 
 non-stack binding]
  37674:  177777 177777 [padding]
  37676:     122  66204 [padding]
  37700:     122  23570 [padding]
  37702:       0      0 NIL
  37704:  177774      4 
  37706:  177774     12 
NC.AssignTitle


Basic frame at   37574
  37572:     122  23570 Window {WINDOW}#122,23570
  37574:  100000  37572 

Frame xtn at   37576, frame name= NC.TEditMenuFn
  37576:  140001  37547 [USE= 1, X,  alink]
  37600:     132  75250 [fn header]
  37602:   37632    153 [next, pc]
  37604:   37634    314 [nametable]
  37606:   37574  37546 [blink, clink]
  37610:     134  33024 "local" NC.AssignTitle
  37612:  177777 177777 LISPXHIST [unbound]
  37614:  177777 177777 *RESETFORMS* [unbound]
  37616:  177777 177777 RESETSTATE [unbound]
  37620:  177777 177777 [fvar LISPXHIST  not looked up]
  37622:  177777 177777 [padding]
  37624:  177777 177777 [padding]
  37626:     122  23570 [padding]
  37630:  177776      0 
NC.TEditMenuFn


Basic frame at   37532
  37526:     134  11326 *FORM* (NC.TEditMenuFn (QUOTE {
WINDOW}#122,23570))
  37530:     132  33264 *local* NC.TEditMenuFn
  37532:  100000  37526 

Frame xtn at   37534, frame name= \EVALFORM
  37534:  140001  37431 [USE= 1, X,  alink]
  37536:      57   2114 [fn header]
  37540:   37572    504 [next, pc]
  37542:   37530  25300 [nametable]
  37544:   37532  37430 [blink, clink]
  37546:  177777 177777 local [unbound]
  37550:  177777 177777 local [unbound]
  37552:  177777 177777 local [unbound]
  37554:       0      0 *ARGVAL* NIL
  37556:       0      0 *TAIL* NIL
  37560:     132  33264 *FN* NC.TEditMenuFn
  37562:  177774      6 [padding]
  37564:     134  11326 [padding]
  37566:  177774     12 
  37570:       0   4345 *ARGVAL*
\EVALFORM


Basic frame at   37414
  37410:     134  11326 *local* (NC.TEditMenuFn (QUOTE {
WINDOW}#122,23570))
  37412:     122  62204 *local* 
#<Process NC.TEditMenuFn/122,62204>
  37414:  100000  37410 

Frame xtn at   37416, frame name= \MAKE.PROCESS0
  37416:  140001  14015 [USE= 1, X,  alink]
  37420:      61  75400 [fn header]
  37422:   37526   1111 [next, pc]
  37424:       0      0 [nametable]
  37426:   37414  14014 [blink, clink]
  37430:     134  11326 %#FORM# (NC.TEditMenuFn (QUOTE {
WINDOW}#122,23570))
  37432:     122  62204 *CURRENT-PROCESS* 
#<Process NC.TEditMenuFn/122,62204>
  37434:       0   1527 HELPFLAG BREAK!
  37436:      16      0 \CURRENTDISPLAYLINE 0
  37440:      16     17 \#DISPLAYLINES 15
  37442:     146 175200 \LINEBUF.OFD 
#<IO Linebuffer Stream/146,175200>
  37444:     174  74714 *READTABLE* 
#<ReadTable INTERLISP/174,74714>
  37446:     174  67740 \PRIMTERMTABLE {TERMTABLEP}#174,
67740
  37450:     174  70000 \PRIMTERMSA {CHARTABLE}#174,
70000
  37452:     146 175700 TtyDisplayStream 
#<Output Display Stream/146,175700>
  37454:       0      0 SI::*RESETFORMS* NIL
  37456:       0    114 \INTERRUPTABLE T
  37460:     122  23000 \TTYWINDOW {WINDOW}#122,23000
  37462:       0      0 READBUF NIL
  37464:     146 175000 \TERM.OFD 
#<Output Display Stream/146,175000>
  37466:     146 175000 *STANDARD-OUTPUT* 
#<Output Display Stream/146,175000>
  37470:     146 175200 *STANDARD-INPUT* 
#<IO Linebuffer Stream/146,175200>
  37472:       0      0 "local" NIL
  37474:       0      0 "local" NIL
  37476:  177777 177777 \INTERRUPTABLE [unbound]
  37500:  177777 177777 RESETSTATE [unbound]
  37502:   23070      1 [fvar HELPFLAG  on stack]
  37504:  177777 177777 [fvar \LINEBUF.OFD 
 not looked up]
  37506:   23100      1 [fvar *READTABLE*  on stack]
  37510:  177777 177777 [fvar \PROC.KILLME 
 not looked up]
  37512:   23102      1 [fvar \PRIMTERMTABLE  on stack]
  37514:   23104      1 [fvar \PRIMTERMSA  on stack]
  37516:   62246    122 [fvar *DRIBBLE-OUTPUT* 
 non-stack binding]
  37520:      56  10720 [padding]
  37522:     122  62204 [padding]
  37524:  177754     44 
\MAKE.PROCESS0


Basic frame at   14000
  14000:  100000  14000 

Frame xtn at   14002, frame name= T
  14002:  140011     13 [USE= 9, X,  alink]
  14004:      61 103460 [fn header]
  14006:   14026     62 [next, pc]
  14010:       0      0 [nametable]
  14012:   14000     12 [blink, clink]
  14014:       0      0 [padding]
  14016:       0      0 [padding]
  14020:       0      0 NIL
  14022:       0      0 NIL
  14024:       0    114 T
T
```

**Comment by fghalasz** [comment]
Fixed in PR#12 in Notecards repo.

Error was caused by using (WINDOWSPROP W 'TEXTOBJ) on TEdit cards, which is no longer supported under TEdit 4th round.  Changed to to (AND (TEDITWINDOWP W)(TEXTSTREAM W)).

**Comment by rmkaplan** [comment]
TEXTSTREAM now also takes a NOERROR flag, returns NIL instead of error if that’s set.  So maybe you could just say (TEXTSTREAM W T)
